distrobuilder/scripts.go

```diff
@@ -0,0 +1,199 @@
+package main
+
+var preVM = `
+#!/bin/sh
+set -eux
+
+# default imagesize = 2252*1024**2 = 2.2G (the current size we ship)
+IMAGESIZE=2361393152
+DISK_IMAGE="%s"
+MOUNTPOINT="%s"
+CACHEDIR=%s
+FS_LABEL=rootfs
+ROOTFS_DEV_MAPPER=
+LOOP_DEVICE=
+
+create_empty_disk_image() {
+    # Prepare an empty disk image
+    dd if=/dev/zero of="$1" bs=1 count=0 seek="${IMAGESIZE}"
+}
+
+create_partitions() {
```

Comment:
So I think we'll want all our images to look like:
 - GPT partition table only
 - First partition 100MB vfat EF00 (/boot/efi)
 - Second partition is / with configurable filesystem and size

A bunch of the other bits in those scripts feel like they should be either file generators or be scripts in the VM definition. I would expect distrobuilder to take care of creating the image and partitions, formatting them and mounting them where appropriate. It should then have a generator that gets us a basic fstab based on the partitions `blkid` and should expose the block device paths of the EFI and root partitions through environment so the scripts can handle it.

Reply:
@stgraber just a thought here, would it be possible to label the partition in such a way that we could detect when initialising a VM instance and resize the correct filesystem after unpacking it onto its block device/file so that its the same size as the volume? The ubuntu images do this with cloud-init I believe, but would be good if this works for non-cloudinit users too.

Reply:
Yeah, I think the rootfs label is what's used for that. We'll definitely want to make sure that our cloud-init enabled images do the right thing, for the rest, we could add a basic systemd unit in distrobuilder which would achieve the same, at least on ext4.